upup/pkg/fi/cloudup/awstasks/autoscalinggroup_lifecyclehook.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awstasks
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/awsup"
+	"k8s.io/kops/upup/pkg/fi/cloudup/cloudformation"
+	"k8s.io/kops/upup/pkg/fi/cloudup/terraform"
+)
+
+// +kops:fitask
+type AutoscalingLifecycleHook struct {
+	ID        *string
+	Name      *string
+	Lifecycle *fi.Lifecycle
+
+	AutoscalingGroup    *AutoscalingGroup
+	DefaultResult       *string
+	HeartbeatTimeout    *int64
+	LifecycleTransition *string
+
+	Tags map[string]string
```

Comment:
hooks don't support tags so this field can be removed

Reply:
👍 done